File: rzops-web/src/lib/types/audit_log.ts
export interface AuditLogResponse {
  id: string;
  actor_id: string | null;
  action: string;
  resource_type: string | null;
  resource_id: string | null;
  ip_address: string | null;
  user_agent: string | null;
  actor_email: string | null;
  extra_data: unknown;
  created_at: string;
}

export interface AuditLogListResponse {
  data: AuditLogResponse[];
  count: number;
}

export interface ListAuditLogsQuery {
  action?: string;
  resource_type?: string;
  actor_id?: string;
  created_from?: string;
  created_to?: string;
  page?: number;
  per_page?: number;
}

File: rzops-api/domain/src/models/audit_log.rs
use chrono::{DateTime, Utc};
use serde_json::Value;
use uuid::Uuid;

/// AuditLog entity — maps to cmdb_audit_log.
#[derive(Debug, Clone)]
pub struct AuditLog {
    pub id: Uuid,
    pub actor_id: Option<Uuid>,
    pub action: String,
    pub resource_type: String,
    pub resource_id: Option<Uuid>,
    pub ip_address: Option<String>,
    pub user_agent: Option<String>,
    pub extra_data: Value,
    pub created_at: DateTime<Utc>,
}
